Readme.txt
Software Engineering Course Project. We worked in a team of four student developers. The original project was hosted on my teammate's GitHub. We worked on weekly sprints. Switched roles between Scrum master, developer, and tester biweekly. LinkedIn command-line app. 

history.txt
commit 4b07b4a4530d2ffd0825d38ecc334fd1a3aa6093
Merge: d832ae2 b11cd60
Author: tyler-smith-9906 <46972768+tyler-smith-9906@users.noreply.github.com>
Date:   Sun Nov 8 21:38:07 2020 -0500

    Merge pull request #6 from snrampersad/branch9
    
    Branch9

commit b11cd6059efed526fe5242067eb03beaa579352d
Author: tyler-smith-9906 <46972768+tyler-smith-9906@users.noreply.github.com>
Date:   Sun Nov 8 21:27:39 2020 -0500

    Fixed issues with removed jobs and messages

commit c50cee772a2d2f3d2aa842c4986ba8e80c77a935
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 20:14:44 2020 -0500

    friends

commit 103f05712113810c72831b7f937604e256eac8ca
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 20:14:25 2020 -0500

    friends notification test

commit a2b938ff921e586f361cbce5d1852b2fd6e914c6
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 18:00:55 2020 -0500

    notification

commit 0f86fd1fa092e59d48451ecdd43ace7bab3b2531
Author: Byte <bytechase@gmail.com>
Date:   Sun Nov 8 14:24:38 2020 -0500

    Added Testing for newFriend Function

commit 77315de08de58ac3daeb338e36b2a75237d25067
Author: Byte <bytechase@gmail.com>
Date:   Sun Nov 8 13:34:05 2020 -0500

    Adding False Assert

commit a05250d45499ec9d39647b71dbbc3f7f9d726cda
Author: Byte <bytechase@gmail.com>
Date:   Sun Nov 8 13:32:44 2020 -0500

    Fixed Brackets

commit 2ad1b4ce9e8fe3be9e1a550ad3e5fe3e4f14972a
Merge: 884fdc5 02901e8
Author: Byte <bytechase@gmail.com>
Date:   Sun Nov 8 13:31:59 2020 -0500

    Merge branch 'branch9' of https://github.com/snrampersad/InCollege into branch9

commit 884fdc5ccebdb6a14418d3a41181a048f19be3a8
Author: Byte <bytechase@gmail.com>
Date:   Sun Nov 8 13:31:08 2020 -0500

    Test System Remind Student to Apply for Job
    
    Testing to see if system reminds student to apply for a job

commit 02901e8640fc7d5b1f1d7b1dee55463a846ad66d
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 13:17:40 2020 -0500

    DB null case added

commit 2d850f833abb370e3f4223771c08b70823116c3b
Merge: 1dab27f 11043b2
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 12:00:54 2020 -0500

    Merge branch 'branch9' of https://github.com/snrampersad/InCollege into branch9

commit 1dab27f9bcb3768bebcf5ec6ef86e6c423da7295
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 12:00:51 2020 -0500

    test

commit 11043b2c305e8ec947563bfdec52761e13c01352
Author: Byte <bytechase@gmail.com>
Date:   Sun Nov 8 11:43:57 2020 -0500

    showNotAppliedJobs test

commit d83b2eb051725147d6823eade539e7e29ee92e10
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 11:21:46 2020 -0500

    jobs

commit 9a63dcc41472d9db4404e467880fdd7069a09822
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 10:20:56 2020 -0500

    friend

commit 8be729179059d711c5fb3d3116c0c5ed9e7f62ee
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 10:19:04 2020 -0500

    friend notif

commit 14feaaf3e0137071aadb912e4185fbf02d159354
Merge: ef5d7fc fda523c
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 10:17:17 2020 -0500

    Merge branch 'branch9' of https://github.com/snrampersad/InCollege into branch9

commit ef5d7fc0091769882260a6b45265172260f12763
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sun Nov 8 10:16:45 2020 -0500

    notifies new friend

commit fda523cd259076c61ac4da101693a658cccca6a5
Author: Byte <bytechase@gmail.com>
Date:   Sun Nov 8 09:56:03 2020 -0500

    Test for Show Not Applied Jobs
    
    Also removed send Messages

commit abad9cce4082ee19e7ef8f4cfb9b55f9ce9289bf
Author: Serena Rampersad <50842980+snrampersad@users.noreply.github.com>
Date:   Sun Nov 8 00:13:09 2020 -0500

    Testing2 updated

commit 00d77ee3792c24b6eca35b5af0324dc791f328d4
Author: Serena Rampersad <50842980+snrampersad@users.noreply.github.com>
Date:   Sat Nov 7 23:25:38 2020 -0500

    Fixed Testing.py

commit 12314d6276855ab45d985293182ea3512a303c8d
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sat Nov 7 11:23:20 2020 -0500

    new friend suggestion

commit a94929c1c7ec739a1a1ad193889f68f8e4d7cb76
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Sat Nov 7 11:21:05 2020 -0500

    Remind student to apply if not done

commit 615190f9ecd4e7b979219bc611bf2d50b1402307
Author: tyler-smith-9906 <46972768+tyler-smith-9906@users.noreply.github.com>
Date:   Fri Nov 6 22:34:25 2020 -0500

    Week 9
    
    Adds support for create profile notifs, message notifs, applied job count, and jobs no longer applied to

commit d832ae221c8e66b5f6c2717cd0f49754be95dbac
Merge: d061614 cdc4070
Author: tyler-smith-9906 <46972768+tyler-smith-9906@users.noreply.github.com>
Date:   Sun Nov 1 22:18:20 2020 -0500

    Merge pull request #5 from snrampersad/epic-7
    
    Epic 7

commit cdc407046a69ea2065cdd7502caa29639932a33e
Author: Akash Singh <aki.singh09@gmail.com>
Date:   Sun Nov 1 16:33:14 2020 -0500

    Plus functionality test and previous epic test updated

commit 0e33fdcf6520a0e50d436d89da85fc8d4be76c01
Author: Byte <bytechase@gmail.com>
Date:   Fri Oct 30 16:13:26 2020 -0400

    Finished Testing of sending Messages

commit 377ed3c5f4143b6044838fb9b8532f493711fc19
Merge: 3077dab 0333070
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Fri Oct 30 11:10:43 2020 -0400

    Merge branch 'epic-7' of https://github.com/snrampersad/InCollege into epic-7

commit 3077dab254a9a841fa52ac5e8f8eb6b5dbbfc9fd
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Fri Oct 30 11:10:38 2020 -0400

    initialize table with membership

commit 0333070bd6d99e01fef44c0c28f10e625bdeb90e
Author: Byte <bytechase@gmail.com>
Date:   Fri Oct 30 10:40:11 2020 -0400

    Removed epic4sram_test.py

commit ed0f73fc7a0992d1914e970de7d13ed7ca062f5a
Author: Siddhant Raman <sraman0302@gmail.com>
Date:   Thu Oct 29 17:51:15 2020 -0400

    Plus members can message on messaging option

commit 3dbc5ce4e435be37d5207aa8781235a4ab39975b

main.py
import notifs
import profile
import guestControls
import user
import options
from notifs import newFriend, newJob
import sys
import login
import friend
from sqlite3.dbapi2 import OperationalError
import sqlite3
import courses
import notifs
import api
# Team Minnesota
# Developers: Siddhant Raman, Tyler Smith
# Epic 3

# API Calls
api.studentAccountsAPI()
api.jobsAPI()
api.trainingAPI()
api.myCollegeJobsAPI()
api.myCollegeProfilesAPI()
api.myCollegeUsersAPI()
api.appliedJobsAPI()
api.savedJobsAPI()
api.myCollegeTraining()

x = 0  # this is the option that the user picks from the main menu
loggedIn = 0  # keeps track of user status
lang = 1  # keeps track of the language
currentUser = "Temp"

print('Welcome to InCollege!')
print()

while x != 19:

    if loggedIn == 1:
        hold = notifs.notifyMessage(currentUser)
        print("The following jobs that you have applied to have been removed")
        print(notifs.beenRemoved(currentUser))
        print("You have: " + str(hold) + " messages")
    if loggedIn == 0:
        print('STUDENT SUCCESS STORY: Tom was a senior in college, about to graduate. He was about to enter the \'real world\' and his soft skills were lacking.')
        print('Tom wasn\'t sure how to build his network of connections. He decided to join InCollege and meet new people. With the help of ')
        print('InCollege, Tom was able to connect with other college students who had strong soft skills and learn from them. He was also able to')
        print('learn about new job listings in his dream location that led to a promised job following graduation!\n')
        print('Find out why you should join InCollege! To see more, type 0.')

        print()
        print('Account Services:')
        print('1. Login')
        print('2. Create an Account')
    else:
        print("Current User: ", currentUser)
        name = newFriend(currentUser)
        print("We have ", name, " in the system. Try adding them to your network!")
        jobs = newJob(currentUser)
        print("We have ", jobs, " job in the system. Try applying to your profile!")

        print('Your language is set to:')
        if lang == 1:
            print("English")
        else:
            print("Spanish")
        print('Account Services:')
        print('1. Login (You are logged in.)')
        print('1. Create an Account (You are logged in.)')

    print()

    print('The following services are offered:')
    if loggedIn == 0:
        print(
            '3. Search for Jobs (Service unavailable. Please log in to use this service.)')
    else:
        print('3. Search for Jobs')
    print('4. Search for People (Send Friend Request)')
    print('5. Learn New Skills')

    if loggedIn == 0:
        print('6. Connect with People (Service unavailable. Please log in to use this service.)')
    else:
        print('6. Connect with People')

    if loggedIn == 0:
        print('7. Post a Job (Service unavailable. Please log in to use this service.)')
    else:
        print('7. Post a Job')
    if loggedIn == 0:
        print('8. Help (Service unavailable. Please log in to use this service.)')
    else:
        print('8. Help')

    print("9. Useful Links")
    print("10. inCollege Links")

    if loggedIn == 0:
        print('11. Profiles (Service unavailable. Please log in to use this service.)')
    else:
        print('11. Profiles')

    if loggedIn == 1:
        print('12. Show My Network')

    if loggedIn == 1:
        if friend.pending_count(currentUser) > 0:
            print('13. Pending sent/received friend requests (You have (' +
                  str(friend.pending_count(currentUser)) + ') pending friend requests!!)')
        else:
            print('13. Pending sent/received friend requests')

    if loggedIn == 1:
        print('14. Applied Jobs')
    if loggedIn == 1:
        print('15. Saved Jobs')
    if loggedIn == 1:
        print('16. Delete a Job Posting')
    if loggedIn == 1:
        print('17. Messages')

    if loggedIn == 1:
        print('18. Plus')
    if loggedIn == 1:
        print('19. Incollege Learning')

    print()
    print('20. Exit')
    print()

    option = input("Please enter a main menu option or enter 20 to exit: ")
    print()

    if(option == '0'):
        x = 0
        print('Video is now playing.')
        print()
        # give user the option to return to the main menu
        print('Would you like to return to the main menu?')
        leave = input(
            'Please enter a "y" to return to the main menu or enter a "n" to exit the program: ')  # y or n

        if(leave == 'n'):
            sys.exit()  # will exit the program

        print()  # formatting

    if(option == '1'):
        temp = options.loginFunc(loggedIn)
        if temp != 1 and temp != 0:
            x = 0
            currentUser = temp
            show = options.showNotAppliedJobs(currentUser)
            if (show == 0):

                print("Remember – you're going to want to have a job when you graduate. Make sure that you start to apply for jobs today!")
        if temp != 0:
            loggedIn = 1

    if (option == '2'):
        temp = options.signup(loggedIn)
        if temp != 1 and temp != 0:
            x = 0
            currentUser = temp
        if temp != 0:
            loggedIn = 1

    if (option == '3'):
        x = 3
        if (loggedIn == 1):
            if options.job_search(currentUser) == False:
                continue  # call job search function
            print()
        else:
            print("Log in to use this feature")

    if (option == '4'):
        if (loggedIn == 1):
            x = 4
            options.find_people(currentUser)
            print()
        else:
            print("Log in to use this feature")

    if (option == '5'):
        x = 5
        options.learn_skills()  # call learn function
        print()
    if(option == '6'):
        if (loggedIn == 0):
            print('You are not logged in.')
            print()
        else:
            x = 6
            options.connect_people(currentUser)  # Call Connect Function
        print()

    if (option == '7'):
        if(loggedIn == 0):
            print('You are not logged in.')
            print()
        else:
            x = 7
            title = input('Enter job title: ')
            description = input('Enter job description: ')
            employer = input('Enter name of employer: ')
            location = input('Enter job location: ')
            salary = input('Enter job salary: ')
            options.postJob(title, description, employer, location, salary)
            print()
    if(option == '8'):
        print()
        if(loggedIn == 0):
            print("You are not logged in.")
        else:
            x = 8
            firstName = input("Enter your first name: ")
            lastName = input("Enter last name: ")
            options.help_page(firstName, lastName)
    if(option == '9'):
        print()
        x = 9
        options.useful_link(loggedIn)
    if(option == '10'):
        x = 11
        options.imp_link(currentUser)
        if currentUser != "Temp":
            x = 10
            lang = guestControls.getLang(currentUser)
    if(option == '11'):
        if loggedIn == 1:
            x = 11
            options.profilePage(currentUser)
        else:
            print("Log in to use this feature")
    if(option == '12'):
        if loggedIn == 1:
            x = 12
            options.showMyNetwork(currentUser)
        else:
            print("Log in to use this feature")
    if(option == '13'):
        if loggedIn == 1:
            x = 13
            friend.pendingDisplay(currentUser)
        else:
            print("Log in to use this feature")
    if(option == '14'):
        if loggedIn == 1:
            x = 14
            applied = options.showAppliedJobs(currentUser)
            show = options.showNotAppliedJobs(currentUser)
        else:
            print("Log in to use this feature")
    if(option == '15'):
        if loggedIn == 1:
            x = 15
            print('Saved Jobs')
        else:
            print("Log in to use this feature")

    if(option == '16'):
        if loggedIn == 1:
            x = 16
            options.getJobPostings(currentUser)
        else:
            print("Log in to use this feature")

    if(option == '17'):
        if loggedIn == 1:
            x = 17
            options.seeInbox(currentUser)
        else:
            print("Log in to use this feature")
    if(option == '18'):
        print()
        if loggedIn != 1:
            print("Log in to use feature")
        else:
            options.plus(currentUser)
    if(option == '19'):
        print()
        if loggedIn != 1:
            print("Log in to use feature")
        else:
            options.selectCourse(currentUser)

    if (option == '20'):
        x = 19
        print('Have a great day!')
        # this is will exit the program
    if (int(option) < 0 or int(option) > 19):
        print("Please Make a Valid Choice\n")
